base_layer/core/tests/tests/header_sync.rs

```diff
@@ -0,0 +1,294 @@
+};
+
+static EMISSION: [u64; 2] = [10, 10];
+
+#[allow(clippy::too_many_lines)]
+#[tokio::test(flavor = "multi_thread", worker_threads = 1)]
+async fn test_header_sync() {
+    // Create the network with alice node and bob node
+    let network = Network::LocalNet;
+    let temp_dir = tempdir().unwrap();
+    let key_manager = create_test_core_key_manager_with_memory_db();
+    let consensus_constants = ConsensusConstantsBuilder::new(network)
+        .with_emission_amounts(100_000_000.into(), &EMISSION, 100.into())
+        .build();
+    let (initial_block, _) = create_genesis_block(&consensus_constants, &key_manager).await;
+    let consensus_manager = ConsensusManagerBuilder::new(network)
+        .add_consensus_constants(consensus_constants)
+        .with_block(initial_block.clone())
+        .build()
+        .unwrap();
+    let (alice_node, bob_node, consensus_manager) = create_network_with_2_base_nodes_with_config(
+        MempoolServiceConfig::default(),
+        LivenessConfig {
+            auto_ping_interval: Some(Duration::from_millis(100)),
+            ..Default::default()
+        },
+        P2pConfig::default(),
+        consensus_manager,
+        temp_dir.path().to_str().unwrap(),
+    )
+    .await;
+    let shutdown = Shutdown::new();
+    let (state_change_event_publisher, _) = broadcast::channel(10);
+    let (status_event_sender, _status_event_receiver) = watch::channel(StatusInfo::new());
+
+    // Alice needs a state machine for header sync
+    let mut alice_state_machine = BaseNodeStateMachine::new(
+        alice_node.blockchain_db.clone().into(),
+        alice_node.local_nci.clone(),
+        alice_node.comms.connectivity(),
+        alice_node.comms.peer_manager(),
+        alice_node.chain_metadata_handle.get_event_stream(),
+        BaseNodeStateMachineConfig::default(),
+        SyncValidators::new(MockValidator::new(true), MockValidator::new(true)),
+        status_event_sender,
+        state_change_event_publisher,
+        RandomXFactory::default(),
+        consensus_manager.clone(),
+        shutdown.to_signal(),
+    );
+
+    // Add 1 block to bob's chain
+    let block_1_bob = append_block(
+        &bob_node.blockchain_db,
+        &initial_block,
+        vec![],
+        &consensus_manager,
+        Difficulty::from_u64(3).unwrap(),
+        &key_manager,
+    )
+    .await
+    .unwrap();
+    assert_eq!(block_1_bob.height(), 1);
+    assert_eq!(bob_node.blockchain_db.get_height().unwrap(), 1);
+
+    // Alice attempts header sync, still on the genesys block, headers will be lagging
+    let event = sync_headers(&mut alice_state_machine, &alice_node, &bob_node).await;
+    // "Lagging"
+    match event.clone() {
+        StateEvent::HeadersSynchronized(val, sync_result) => {
+            assert_eq!(val.claimed_chain_metadata().height_of_longest_chain(), 1);
+            assert_eq!(val.claimed_chain_metadata().accumulated_difficulty(), 4);
+            assert_eq!(sync_result.headers_returned, 1);
```

Comment:
Nit: This is probably fine as is here. I would have preferred to test these values in unit tests, then we don't have to add code to allow testing of values. These values are only used by these tests.

Reply:
Agreed that it is probably fine